### ISSUE TITLE
[MNG-5555] @parameter expression="${mojoExecution.lifecyclePhase}" does not work in 3.1.1

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#6782]
**[Martin Gainty](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mgainty)** opened **[MNG-5555](https://issues.apache.org/jira/browse/MNG-5555?redirect=false)** and commented

Maven-core:3.1.1

```java
/**
  * @parameter expression="${mojoExecution}"
*/
@Parameter(defaultValue = "${mojoExecution}")
private org.apache.maven.plugin.MojoExecution execution;
public org.apache.maven.plugin.MojoExecution getExecution(){ return execution; }
/**
 * @parameter expression="${mojoExecution.lifecyclePhase}"
 */
@Parameter(defaultValue = "${mojoExecution.lifecyclePhase}")
private String lifecyclePhase;
public org.apache.maven.model.Plugin plugin=new org.apache.maven.model.Plugin();
public String getLifecyclePhase()
{
	if(lifecyclePhase==null)
	{
		plugin.setGroupId("org.apache.maven.plugins");
		plugin.setArtifactId("maven-compiler-plugin");
		plugin.setVersion("3.1");
		String goal="compile";
		String executionId="executionId";
		System.out.println("CompilerMojoTestCase::getLifecyclePhase goal="+goal+" executionId="+executionId);
		org.apache.maven.plugin.MojoExecution mojo=new org.apache.maven.plugin.MojoExecution((org.apache.maven.model.Plugin)plugin, goal, executionId );
		lifecyclePhase= mojo.getLifecyclePhase();
	}
	return lifecyclePhase;
}
```

produces:

```
CompilerMojoTestCase::getLifecyclePhase goal=compile executionId=executionId
CompilerMojoTestCase::testCompilerBasic LINE 143 getLifeyclePhase()=     null
```

I am using 3.1.1 jars top to botton
why does lifecyclePhase always produce null?

---

**Affects:** 3.1.0-alpha-1

**Comment by jira-importer** [comment]
**[Martin Gainty](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mgainty)** commented

status changed to 'resolved' when implementing JDK 1.7.0_45
The bug is only reproducible with JDK 1.7.0_25
This is a JVM 1.7.0_25 specific bug and Not a maven bug
Martin Gainty
31 Dec 2013

**Comment by jira-importer** [comment]
**[Anders Hammar](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=afloom)** commented

Changing resolution to Not A Bug.
